src/components/Link/Link.mdx

```diff
@@ -36,9 +36,10 @@ import * as LinkStories from "./Link.stories";
 
 ## Accessibility
 
-The `Link` component should be used for navigation. If an `onClick` user
-interface action is required, a `Button` component should be used instead. The
-`Link` component renders an `<a>` element with the `href` attribute.
+The `Link` component renders an `<a>` element with the `href` attribute, so the
+`Link` component should be used for navigation. If an `onClick` user interface
+action is required, a `Button` component should be used instead. In simple
+terms, a link goes somewhere and a button does something.
```

Comment:
I think we should mention that it still is possible to pass an `onClick` to the link component because there are some very specific use cases and it is also a native HTML pattern that is allowed (but the HTML attribute is `onclick` all lowercase). We just really urge against it unless they know what they are doing.